django_material_demo/cms/polls/views/question.py

```diff
@@ -310,28 +307,33 @@ def get_object_data(self):
         thumbnail_name = thumbnail_name.verbose_name.title()
         for item in super().get_object_data():
             if item[0] == thumbnail_name:
-                # Skip if no image
                 if item[1]:
-                    # TODO: replace with template
-                    image_html = mark_safe(
-                        f"<img class='thumbnail' src='{item[1].url}' "
-                        f"alt='{item[1].name}'>")
+                    ctx = {'attrs': {
+                        'class': 'thumbnail',
+                        'src': item[1].url,
+                        'alt': item[1].name
+                    }}
+                    image_html = render_to_string('data/img.html', ctx)
                     yield (item[0], image_html)
                 else:
                     yield (item[0], 'None')
             else:
                 yield item
 
         attachments = question.attachment_set.all()
-        if len(attachments):
-            # TODO: replace with template
-            attachments = [
-                mark_safe(f"<a href='{x.file.url}'>{x.file.name}</a>")
-                for x in attachments]
-            html_list = get_html_list(attachments)
-            yield ('Attachments', html_list)
-        else:
-            yield ('Attachments', 'None')
+        attachment_links = []
+        for attachment in attachments:
+            ctx = {
+                'content': attachment.file.name,
+                'attrs': {
+                    'href': attachment.file.url,
+                    'download': True
+                },
+            }
+            attachment_links.append(render_to_string('data/a.html', ctx))
+        ctx = {'items': attachment_links}
+        html_list = render_to_string('data/ul.html', ctx)
```

Comment:
This is not how we usually use the template engine. If the template is a collection of data and requires another template file, we specify the other template within the collection template, not rendering it outside and pass the string as part of context data. It should be something like https://stackoverflow.com/questions/5733172/django-how-to-partial-render.

You will need to transform the db data to view data at python code all at once, and then pass a single context to render `data/ul.html`.